pkg/engine/artifacts.go

```diff
@@ -30,111 +30,115 @@ func kubernetesContainerAddonSettingsInit(p *api.Properties) map[string]kubernet
 	}
 	o := p.OrchestratorProfile
 	k := o.KubernetesConfig
```

Comment:
Below is the meat of the matter. We are now simply relying upon the fully constructed addons array to inform how we concatenate the output string that we inject into cloud-init.